cdc/metrics_server.go

```diff
@@ -14,19 +14,42 @@
 package cdc
 
 import (
+	"os"
+	"strconv"
+
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-var etcdHealthCheckDuration = prometheus.NewHistogramVec(
-	prometheus.HistogramOpts{
-		Namespace: "ticdc",
-		Subsystem: "server",
-		Name:      "etcd_health_check_duration",
-		Help:      "Bucketed histogram of processing time (s) of flushing events in processor",
-		Buckets:   prometheus.ExponentialBuckets(0.0001 /* 0.1ms */, 2, 18),
-	}, []string{"capture", "pd"})
+var (
+	etcdHealthCheckDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "ticdc",
+			Subsystem: "server",
+			Name:      "etcd_health_check_duration",
+			Help:      "Bucketed histogram of processing time (s) of flushing events in processor",
+			Buckets:   prometheus.ExponentialBuckets(0.0001 /* 0.1ms */, 2, 18),
+		}, []string{"capture", "pd"})
+
+	goGC = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "ticdc",
+			Subsystem: "server",
+			Name:      "gogc",
+			Help:      "The value of GOGC",
+		})
+)
+
+func init() {
```

Comment:
Find a similar `init` function in tidb util package, I prefer to check `GOGC` env and set it with an explicit function.
Besides seems go runtime has processed `GOGC` env variable

https://github.com/golang/go/blob/170a72e58bd128b421f4b3974fe2a37fd035efdf/src/runtime/mgc.go#L190-L208